sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/FieldBuilder.java

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.search.annotation.FieldIgnore;
+import com.azure.search.annotation.SearchableFieldProperty;
+import com.azure.search.annotation.SimpleFieldProperty;
+import com.azure.search.documents.models.AnalyzerName;
+import com.azure.search.documents.models.DataType;
+import com.azure.search.documents.models.Field;
+import com.azure.search.documents.models.GeoPoint;
+import com.azure.search.documents.models.SearchableField;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+import java.util.stream.Collectors;
+
+/**
+ * Helper to covert model class to search field {@link Field}.
+ */
+public class FieldBuilder {
+    private static final int MAX_DEPTH = 10;
```

Comment:
This is kind of protection to avoid deep calling. We just recommend customer to redesign their index if it is call path is too deep. Like complexField 1-> 2 -> 3 ... -> 10 -> 11 (Fail here). I don't have reference for the number(willing to change). Also, it is just a good to have validation.

Reply:
Depth validation should be left to the service. The depth check done by the REST API can vary based on pricing tier, so having a hard-coded value here is likely to cause breakage for at least some customers, unless you pick a very large value.

If you're checking for recursive types, then unbounded stack growth isn't possible. If you're just concerned about a very deep tree of other types, pick a much larger value, like 10000, that is likely to always be greater than the REST API's own limits.

Reply:
Change it to 10000